### ISSUE TITLE
OrthoGrad Support

### ISSUE THREAD

**Issue by HydrogenBombaklot** [stockeh/mlx-optimizers#65]
OrthoGrad, along with StableMax, are introduced in this repo: https://github.com/LucasPrietoAl/grokking-at-the-edge-of-numerical-stability

Their [paper](https://arxiv.org/abs/2501.04697) discusses how Stablemax and OrthoGrad combined prevent Softmax collapse, prevent naive loss minimization, and ultimately lead to much improved generalization.

I have been advised that dropping in StableMax to replace Softmax in MLX is very easy. I'm wondering how we can implement OrthoGrad into MLX as well.

**Comment by stockeh** [comment]
@HydrogenBombaklot similar to #61 and [Muon](https://github.com/stockeh/mlx-optimizers/blob/main/mlx_optimizers/muon.py) (for 1D params), I wonder if there's a better way to add wrapping. Instead of passing a base optimizer, e.g.
```python
class OrthoGrad(torch.optim.Optimizer):
    def __init__(self, params, base_optimizer_cls=torch.optim.SGD, **base_optimizer_args):
```
Do you have any thoughts on this? I don't see it being a challenge to add otherwise.

**Comment by HydrogenBombaklot** [comment]
HI @stockeh I have no thoughts on this. I just read the paper and thought it would be great to integrate into MLX. I don't have the technical capability to understand your code, much less contribute anything to it.
